### PR TITLE
[Http] expose necessary api for multipart

### PR DESCRIPTION
**Pull request by msyyc**
TCGC needs these API exposed

### CODE REVIEW

File: packages/http/src/index.ts [hunk 1]
```diff
@@ -7,7 +7,14 @@ export * from "./decorators.js";
...
 export * from "./parameters.js";
-export { getHttpFileModel, isHttpFile, isOrExtendsHttpFile } from "./private.decorators.js";
+export {
+  HttpPart,
+  HttpPartOptions,
+  getHttpFileModel,
+  getHttpPart,
+  isHttpFile,
+  isOrExtendsHttpFile,
+} from "./private.decorators.js";
```
Comment (MaryGao) [inline packages/http/src/index.ts:17]: I notice this is in private decorator, should we move them out of private if we have public usage?

**Review by timotheeguerin — CHANGES_REQUESTED** [review]
Can you explain more why this needs to be exposed. There is a reason they are internal, the public info is via the http operation

**Comment by msyyc** [comment]
> Can you explain more why this needs to be exposed. There is a reason they are internal, the public info is via the http operation

Please see https://github.com/Azure/typespec-azure/pull/1090/files/223f68e13c989cfafd483f01034afeaeb2b59b66#r1660410950

File: .chronus/changes/expose-necessary-api-for-httppart-2024-5-27-18-23-11.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+changeKind: feature
+packages:
+  - "@typespec/http"
+---
+
+Expose API for HttpPart
```
Comment (timotheeguerin) [inline .chronus/changes/expose-necessary-api-for-httppart-2024-5-27-18-23-11.md:7]:
```suggestion
Expose `getHttpPart` and types functions
```

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**